plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -373,27 +366,32 @@ def _handle_package(repo, type_id, unit_key, metadata, file_path, conduit, confi
         _LOGGER.exception('Error extracting RPM metadata for [%s]' % file_path)
         raise
 
+    # metadata can be None
+    metadata = metadata or {}
+
     model_class = plugin_api.get_unit_model_by_id(type_id)
     update_fields_inbound(model_class, unit_key or {})
     update_fields_inbound(model_class, metadata or {})
 
-    # set checksum and checksumtype
-    if metadata:
-        checksumtype = metadata.pop('checksum_type', verification.TYPE_SHA256)
-        rpm_data['checksumtype'] = verification.sanitize_checksum_type(checksumtype)
-        if 'checksum' in metadata:
-            rpm_data['checksum'] = metadata.pop('checksum')
-            try:
-                with open(file_path) as dest_file:
-                    verification.verify_checksum(dest_file, rpm_data['checksumtype'],
-                                                 rpm_data['checksum'])
-            except verification.VerificationException:
-                raise PulpCodedException(error_code=platform_errors.PLP1013)
-        else:
-            rpm_data['checksum'] = _calculate_checksum(rpm_data['checksumtype'], file_path)
-    else:
-        rpm_data['checksumtype'] = verification.TYPE_SHA256
-        rpm_data['checksum'] = _calculate_checksum(rpm_data['checksumtype'], file_path)
+    with open(file_path) as fp:
+        sums = util.calculate_checksums(fp, [util.TYPE_MD5, util.TYPE_SHA1, util.TYPE_SHA256])
```

Comment:
Does it make any sense to de-duplicate this list as `PULP_AUTO_DEFAULTS` or some better name than that?